src/classes/RD2_EntryFormController.cls

```diff
@@ -314,6 +316,24 @@ public with sharing class RD2_EntryFormController {
     }
 
 
+    /**
+    * @description Retrives Recurring Donation from database
+    * @param recordId Recurring Donation Id
+    * @return npe03__Recurring_Donation__c
+    */
+    private static npe03__Recurring_Donation__c getOldRecurringDonation(Id recordId) {
+        npe03__Recurring_Donation__c oldRd = null;
+
+        if (recordId != null) {
+            List<npe03__Recurring_Donation__c> oldRds = new RD2_QueryService().getRecurringDonationsBase(new Set<Id>{ recordId });
```

Comment:
We have so much logic to rewrite if we ever want to support batch or trigger-based updates 😅

Reply:
I think we're going to be looking at pushing things down into `RD_RecurringDonations` (Domain class) and `RD2_RecurringDonation` (Entity) to move to batch/trigger based updates. Two instances of that Entity class working together should be able to make the logic for comparing RDs look much cleaner.